crates/dojo-lang/src/commands/delete.rs

```diff
@@ -0,0 +1,93 @@
+use std::collections::HashMap;
+
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_semantic::patcher::RewriteNode;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{ast, Terminal, TypedSyntaxNode};
+
+use super::{CommandData, CommandTrait};
+
+#[derive(Clone)]
+pub struct DeleteCommand {
+    data: CommandData,
+    pub components: Vec<smol_str::SmolStr>,
+}
+
+impl DeleteCommand {
+    fn handle_struct(&mut self, db: &dyn SyntaxGroup, query: ast::Arg, expr: ast::Expr) {
+        if let ast::Expr::StructCtorCall(ctor) = expr {
+            if let Some(ast::PathSegment::Simple(segment)) = ctor.path(db).elements(db).last() {
+                let component = segment.ident(db).text(db);
+
+                self.components.push(component.clone());
+                self.data.rewrite_nodes.push(RewriteNode::interpolate_patched(
+                    "
+                    {
+                        let mut calldata = array::ArrayTrait::new();
+                        serde::Serde::serialize(@$ctor$, ref calldata);
+                        IWorldDispatcher { contract_address: world_address \
+                     }.delete_entity(dojo_core::string::ShortStringTrait::new('$component$'), \
+                     $query$, 0_u8, array::ArrayTrait::span(@calldata));
```

Comment:
```suggestion
                     }.delete_entity(dojo_core::string::ShortStringTrait::new('$component$'), $query$);
```

The delete entity api is `fn delete_entity(component: ShortString, query: Query);`

Reply:
you can find it here

https://github.com/dojoengine/dojo/blob/9eaa1a76f74f7268bccee811b31497f55395e1f8/crates/dojo-core/src/world.cairo#L197